src/Cli/dotnet/Commands/Run/VirtualProjectBuildingCommand.cs

```diff
@@ -212,6 +215,13 @@ public override int Execute()
                 var buildRequest = new BuildRequestData(
                     CreateProjectInstance(projectCollection),
                     targetsToBuild: [BuildTarget]);
+
+                // For some reason we need to BeginBuild after creating BuildRequestData otherwise the binlog doesn't contain Evaluation.
```

Comment:
> Should we pull some MSBuild folks in here to verify we're doing things in the correct order?

Good idea. @rainersigwald.

If you would like a simplified self-contained repro, use this:

https://github.com/jjonescz/MSBuildInMemory/commit/599e596453e665fc2eb95eb0a44eccd1e08667ca

that last commit is also the difference between binlog containing evaluation data and not (in the previous commit of the repro app).